From d896ab41aa87e65ac112a5a94515dfc15594a6c9 Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <peleg.vigodny@tandemg.com>
Date: Mon, 13 Jul 2020 18:13:50 +0300
Subject: [PATCH] [SW-15627] openbmc: host: install pcie sw update

Add hl_pcie_sw_update.sh host tool to the openbmc build and deploy it as
part of the HLS release.

This tools will allow updating PCIe switch FW remotely via the BMC.

Change-Id: Ibdeff2f389796ef2d1052aeebf690e1dc6815daf
Signed-off-by: Peleg Vigodny <peleg.vigodny@tandemg.com>
---

diff --git a/package/openbmc/openbmc.mk b/package/openbmc/openbmc.mk
index b788f70..c053a59 100644
--- a/package/openbmc/openbmc.mk
+++ b/package/openbmc/openbmc.mk
@@ -6,6 +6,7 @@
 OBMC_PHOSPHOR_IMAGE = $(OPENBMC_IMAGE_DIR)/$(OBMC_PHOSPHOR_IMAGE_NAME)
 OBMC_PHOSPHOR_FULLCHIP_IMAGE = $(OPENBMC_IMAGE_DIR)/$(OBMC_PHOSPHOR_FULLCHIP_IMAGE_NAME)
 OPENBMC_FRUGEN_PATH = $(OPENBMC_IMAGE_DIR)/$(HLS_EEPROM_FRUGEN)
+OPENBMC_PCIE_SW_UPDATE =  $(OPENBMC_IMAGE_DIR)/hl_pcie_sw_update.sh
 VERSION_FILE := $(OPENBMC_DEPLOY_DIR)/localversion-habanalabs
 
 define SET_VERSION_INFO
@@ -43,6 +44,26 @@
 
 OPENBMC_POST_BUILD_HOOKS += OPENBMC_BUILD_FRUGEN
 
+define OPENBMC_CLEAN_PCIE_SW_UPDATE
+	(cd $(@D); 							\
+		$(OPENBMC_ENV)						\
+		$(OPENBMC_BITBAKE_ENV)					\
+		bitbake -c cleanall hl-pcie-sw-update-native;		\
+	)
+endef
+
+OPENBMC_PRE_BUILD_HOOKS += OPENBMC_CLEAN_PCIE_SW_UPDATE
+
+define OPENBMC_INSTALL_PCIE_SW_UPDATE
+	(cd $(@D); 							\
+		$(OPENBMC_ENV)						\
+		$(OPENBMC_BITBAKE_ENV)					\
+		bitbake hl-pcie-sw-update-native;			\
+	)
+endef
+
+OPENBMC_POST_BUILD_HOOKS += OPENBMC_INSTALL_PCIE_SW_UPDATE
+
 define OPENBMC_INSTALL_IMAGES_CMDS
 	$(INSTALL) -m 0644 -D $(@D)/$(OPENBMC_IMAGE) $(BINARIES_DIR)/$(OPENBMC_IMAGE_NAME).bin
 	$(INSTALL) -m 0644 -D $(@D)/$(OBMC_PHOSPHOR_IMAGE) $(BINARIES_DIR)/$(OBMC_PHOSPHOR_IMAGE_NAME)
@@ -52,6 +73,7 @@
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls-curr.sh $(BINARIES_DIR)/install-hls-curr.sh
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls-legacy.sh $(BINARIES_DIR)/install-hls-legacy.sh
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls-check-dependencies.sh $(BINARIES_DIR)/install-hls-check-dependencies.sh
+	$(INSTALL) -m 0755 -D $(@D)/$(OPENBMC_PCIE_SW_UPDATE) $(BINARIES_DIR)
 	$(INSTALL) -m 0644 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/README.txt $(BINARIES_DIR)/README.txt
 endef
 
diff --git a/scripts/make/hls/Makefile b/scripts/make/hls/Makefile
index de92ed9..92c9d04 100644
--- a/scripts/make/hls/Makefile
+++ b/scripts/make/hls/Makefile
@@ -1,6 +1,7 @@
 export OPENBMC_WORK_DIR = build
 export OPENBMC_ENV_FLAGS = meta-habana/meta-hls/conf
 export OPENBMC_IMAGE_NAME = flash-hls
+export HL_PCIE_SW_UPDATE = hl_pcie_switch_update.sh
 export HLS_EEPROM_FRUGEN = frugen
 export OBMC_PHOSPHOR_IMAGE_BASENAME = obmc-phosphor-image-hls
 export OBMC_PHOSPHOR_IMAGE_NAME = $(OBMC_PHOSPHOR_IMAGE_BASENAME).static.mtd.tar
@@ -48,6 +49,7 @@
 	$(eval HLS_RELEASE_DIR=$(HLS_RELEASE_DIR_BASE)/$(shell date +%Y%m%d-%H%M%S))
 	@sudo mkdir -p "$(HLS_RELEASE_DIR)"
 	@sudo cp -p $(IMAGES_DIR)/$(HLS_PKG_IMG_NAME) $(HLS_RELEASE_DIR)
+	@sudo cp -p $(IMAGE_DIR)/$(HL_PCIE_SW_UPDATE) $(HLS_RELEASE_DIR)
 	@sudo cp -p $(IMAGES_DIR)/$(HLS_EEPROM_FRUGEN) $(HLS_RELEASE_DIR)
 	@echo "$(EMBD_FULL_VER)" | sudo tee $(HLS_RELEASE_DIR)/hl-fw-version.txt
 	@sudo ln -nfs $$(basename $(HLS_RELEASE_DIR)) $(HLS_RELEASE_DIR_BASE)/$(1)-release
